src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -106,7 +114,8 @@ private static Application createEditedPerson(Application applicationToEdit,
         Phone updatedPhone = editApplicationDescriptor.getPhone().orElse(applicationToEdit.getPhone());
         Email updatedEmail = editApplicationDescriptor.getEmail().orElse(applicationToEdit.getEmail());
         Address updatedAddress = editApplicationDescriptor.getAddress().orElse(applicationToEdit.getAddress());
-        Set<Tag> updatedTags = editApplicationDescriptor.getTags().orElse(applicationToEdit.getTags());
+        Set<Tag> updatedTags = editApplicationDescriptor.getTags(applicationToEdit.getTags())
+                .orElse(applicationToEdit.getTags());
```

Comment:
I think you can remove the orElse statement?

Reply:
The orElse statement here is still needed to catch the case where editApplicationDescriptor.getTags() returns an Optional<Set<Tag>> that is empty.

This will happen if the `edit` command isn't used to edit 't/', 'ast/' and 'pt/'. Kind of like a base case. In this case, we will just attempt to get the Tags within the applicationToEdit() object instead (which is basically a reference to the existing application that is being specified to be edited).